### ISSUE TITLE
passing unused option :in to shell causes cast exception

### ISSUE THREAD

**Issue by retrogradeorbit** [epiccastle/spire#144]
spire native:

```
$ SPIREPATH=. spire provisioning/provision.clj
provisioning/provision.clj:46 (shell {:cmd "psql", :in "\\l\n", :print true})
Exception in thread "main" java.lang.ClassCastException: java.lang.String cannot be cast to java.io.InputStream
	at spire.ssh$ssh_exec_proc.invokeStatic(ssh.clj:207)
	at spire.ssh$ssh_exec.invokeStatic(ssh.clj:295)
	at spire.ssh$ssh_exec.invoke(ssh.clj:259)
	at spire.module.shell$shell_STAR_$fn__24449.invoke(shell.clj:126)
	at spire.module.shell$shell_STAR_.invokeStatic(shell.clj:118)
	at spire.module.shell$shell_STAR_.doInvoke(shell.clj:82)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at sci.impl.vars.SciVar.invoke(vars.cljc:325)
	at sci.impl.analyzer$return_call$reify__4565.eval(analyzer.cljc:1149)
...
```

**Comment by retrogradeorbit** [comment]
Parameter is `:stdin` not `:in`. When using `:in`, this option overrides a deeper option passed on to the exec function, replacing a stream object with the naive string. This is through a poorly done `merge` here: https://github.com/epiccastle/spire/blob/master/src/clj/spire/ssh.clj#L294-L298

**Comment by retrogradeorbit** [comment]
Merge be replaced so as not to allow these args to be overridden.

**Comment by retrogradeorbit** [comment]
fixed in 0.1.1
